packages/ui-kit/src/components/shared.types.ts

```diff
@@ -0,0 +1,15 @@
+import type { ErrorFallbackProps } from './ErrorFallback'
+import type { LoaderProps } from './Loader'
+
+/** Shared props for the data components. */
+
+export type DataComponentProps = {
+  /** Optional porps that are used to configure the Loader component. */
+  loaderProps?: LoaderProps
+
+  /** Optional porps that are used to configure the ErrorFallback component. */
+  errorFallbackProps?: ErrorFallbackProps
```

Comment:
I saw this repeated in another file. Should it be combined there? Via interface extension?

```suggestion
  /** Optional props that are used to configure the Loader component. */
  loaderProps?: LoaderProps

  /** Optional props that are used to configure the ErrorFallback component. */
  errorFallbackProps?: ErrorFallbackProps
```